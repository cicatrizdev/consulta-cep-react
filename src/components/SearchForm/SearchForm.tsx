import React, { FormEvent, useState, useContext } from 'react'
import styled from 'styled-components'

import { Store } from '../../contexts/store'
import Button from '../Button/Button'
import InputField from '../InputField/InputField'

const SearchFormContainerWrapper = styled.form`
  background: #0d47a1;
  box-sizing: border-box;
  padding: 2rem 2rem 4rem;
  width: 100%;
  color: white;
`

const SearchFormFieldWrapperLabel = styled.div`
  font-weight: 600;
  margin-bottom: 2rem;
`

const SearchBar = styled.div`
  display: flex;
`

const LabelWrapper = styled.label`
  > * {
    margin-right: 1rem;
  }
`

const SearchFormContainer = () => {
  const ctx = useContext(Store)
  const [zipCode, setZipCode] = useState('')

  const onSubmit = async (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault()

    if (!zipCode) return

    await ctx.action.fetchAddress(zipCode)
  }

  return (
    <SearchFormContainerWrapper onSubmit={onSubmit} data-testid='form-search-container'>
      <SearchFormFieldWrapperLabel>Consultar</SearchFormFieldWrapperLabel>
      <SearchBar>
        <LabelWrapper htmlFor="search-field">
          <span>CEP</span>
          <InputField id="search-field" value={zipCode} onChange={setZipCode} />
        </LabelWrapper>
        <Button />
      </SearchBar>
    </SearchFormContainerWrapper>
  )
}

export default SearchFormContainer
