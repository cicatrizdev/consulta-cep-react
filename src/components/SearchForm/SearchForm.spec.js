import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SearchForm from './SearchForm'

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<SearchForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});