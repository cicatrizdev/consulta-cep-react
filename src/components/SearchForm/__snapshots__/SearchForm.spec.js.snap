// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<form
  className="sc-htpNat drEUuD"
  data-testid="form-search-container"
  onSubmit={[Function]}
>
  <div
    className="sc-bxivhb ebZIbr"
  >
    Consultar
  </div>
  <div
    className="sc-ifAKCX WOpjk"
  >
    <label
      className="sc-EHOje hHgPFw"
      htmlFor="search-field"
    >
      <span>
        CEP
      </span>
      <input
        className="sc-bwzfXH gsZmKd"
        data-testid="search-field"
        id="search-field"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        placeholder="01234-567"
        required={true}
        value=""
      />
    </label>
    <button
      className="sc-bdVaJa faugNV"
    >
      Buscar
    </button>
  </div>
</form>
`;
